File: src/2017/1/day1.rs
use std::fs::File;
use std::path::Path;
use std::io::prelude::*;
use std::vec::Vec;

fn convert_char(c: char) -> u8 {
    c.to_digit(10).unwrap() as u8
}

fn load_file(path: &Path) -> Vec<u8> {
    let mut file = File::open(path).unwrap();
    let mut contents = String::new();
    file.read_to_string(&mut contents).unwrap();
    return contents.chars().map(|x| convert_char(x)).collect::<Vec<u8>>();
}

fn main() {
    let path = Path::new("../resources/day1.txt");
    let nums = load_file(&path);
    let mut total: u32 = 0;
    for (i, digit) in nums.iter().enumerate() {
        if *digit == nums[(i + 1) % nums.len()] {
            total += *digit as u32;
        }
    }
    println!("Total sum = {}", total);
}

File: src/2017/1/day1.py
"""Day 1 solver for 2017 AOC"""

def load_input(filename):
    """Load file and return list of numbers"""
    rtrn = []
    infile = open(filename)
    for line in infile:
        rtrn += [int(x) for x in line]
    return rtrn

def main():
    """Main function"""
    input_list = load_input('../resources/day1.txt')
    total = 0
    for idx, digit in enumerate(input_list):
        next_idx = idx + 1
        if digit == input_list[next_idx % len(input_list)]:
            total += digit
    print(total)
    total_part_2 = 0
    for idx, digit in enumerate(input_list):
        next_idx = idx + (len(input_list) // 2)
        if digit == input_list[next_idx % len(input_list)]:
            total_part_2 += digit
    print(total_part_2)


if __name__ == '__main__':
    main()
